pkg/serverless/trace/inferredspan/inferred_span.go

```diff
@@ -20,6 +25,15 @@ const (
 	coldStartTagKey       = "cold_start"
 )
 
+type InferredSpan struct {
+	Span    *pb.Span
+	IsAsync bool
+}
+
+// Holds all inferred spans with a key of request-id and reduces
+// the risk of multiple invocations manipulating the same global variable
+var InferredSpans = make(map[string]InferredSpan)
```

Comment:
I don't think you need to use this map. There should never be more than one invocation in flight at the same time. I think you could probably use the `currentExecutionInfo` variable that we're already using to keep track of the current trace context.